modules/hal_nordic/nrf_802154/radio/platform/nrf_802154_random_zephyr.c

```diff
@@ -0,0 +1,61 @@
+/*
+ * Copyright (c) 2019 - 2021 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+/*
+ * Based on code written in 2016-2018 by David Blackman and Sebastiano Vigna
+ * (vigna@acm.org)
+ *
+ * To the extent possible under law, the author has dedicated all copyright
+ * and related and neighboring rights to this software to the public domain
+ * worldwide. This software is distributed without any warranty.
+ *
+ * See <http://creativecommons.org/publicdomain/zero/1.0/>.
```

Comment:
The license check seems to be red due to this statement.

Reply:
The problem is that this is not a license listed by the OSI:
https://opensource.org/licenses/alphabetical

Is there any way you can avoid using code under this license?

Reply:
@nashif @MaureenHelm @galak do you know if public domain code is suitable for inclusion?

Reply:
The code under the CC0 license is the PRNG. We don't use zephyr generators, because we need to get a random number from an interrupt context, possibly a zero latency interrupt.

Reply:
Looking at the abovementioned file, it seems it is also under CC0.
```
/*
 * Copyright (c) 2017 Intel Corporation
 *
 * SPDX-License-Identifier: CC0-1.0
 *
 * Based on code written in 2016 by David Blackman and Sebastiano Vigna
 * (vigna@acm.org)
 *
 * To the extent possible under law, the author has dedicated all copyright
 * and related and neighboring rights to this software to the public domain
 * worldwide. This software is distributed without any warranty.
 *
 * See <http://creativecommons.org/publicdomain/zero/1.0/>.
 */
```

Reply:
Actually @e-rk you are right that we [already include](https://github.com/zephyrproject-rtos/zephyr/blob/master/subsys/random/rand32_xoroshiro128.c#L4) this very same code, so I don't see why we couldn't have another copy to be honest.

Reply:
For the time being I have replaced the previous PRNG with a different one taken from OpenThread licensed under BSD 3-clause. Let me know if it's alright.